laborator/lab-10/README.md

```diff
@@ -0,0 +1,384 @@
+> pentru a genera un buffer overflow și pentru a obține rezultatul cerut.
+
+
+## 8. Buffer overflow pentru program scris în cod C
+
+De cele mai multe ori vom identifica vulnerabilități de tip buffer overflow în programe scrise în C.
+Acolo trebuie să vedem ce buffere sunt și care este distanța de la buffer la variabila dorită pentru a putea face suprascrierea. 
+
+> **IMPORTANT** Este important de avut în vedere că distanța între un buffer și o altă variabilă în C poate nu corespunde cu cea "din teren";
+> compilatorul poate face actualizări, reordonări, poate lăsa spații libere între variabile etc.
+
+Pentru exercițiul curent, accesați directorul `8-c-buffer-overflow/` din arhiva de resurse a laboratorului
+și observați codul sursă aferent în C. Pentru cazul în care doriți să nu mai compilați voi codul aveți în arhivă
+și fișierul limbaj de asamblare echivalent și fișierul în cod obiect și fișierul executabil.
+
+Descoperiți diferența între adresa buffer-ului și adresa variabilei, creați un fișier de intrare (numit și `payload`)
+cu care să declanșați overflow-ul și faceți în așa fel încât să fie afișat mesajul *Full of win!*.
+
+> **TIP** Ca să vedeți realitatea "din teren", adică să aflați care este diferența dintre buffer și variabila pe care dorim să o suprascriem,
+> consultați fișierul în limbaj de asamblare echivalent (`do_overflow.asm`), obținut prin asamblarea codului C.
+> În acest fișier puteți afla adresa relativă a buffer-ului față de `ebp` și a variabilei față de `ebp`;
+> urmăriți secvența cuprinsă între liniile `32` și `41`; aveți o mapare între numele variabilei și offset-ul relativ față de `ebp`.
+> Cu aceste informații puteți crea șirul pe care să îl transmiteți ca payload către intrarea standard a programului.
+
+> **NOTE** Dacă doriți să recompilați fișierele rulați:
+> ```Bash
+> make clean && make
+> ```
+
+> **TIP** La fel ca mai sus, ca să transmiteți șirul de intrare pentru program, e recomandat să-l scrieți într-un fișier
+> și apoi să redirectați acel fișier către comanda aferentă > programului.
+> Redirectarea fișierului payload către program se face folosind o comandă precum:
+> ```Bash
+> ./do_overflow < payload
+> ```
+
+
+## 9. Bonus: Stack canary
+
+Pornind de la resursele exercițiului anterior din directorul `8-9-c-buffer-overflow` inspectați fișierul `Makefile`.
+
+> ```Bash
+> cat Makefile
+> ```
+
+Analizați atent opțiunile de compilare. Ce observați? 
+
+Așa cum ați observat în cadrul exercițiului anterior, deși am depășit dimensiunea buffer-ului
+și am suprascris o altă variabilă din program, acesta și-a încheiat execuția în mod normal.
+Acest lucru este nedorit atunci când lucrăm cu buffere, deoarece sunt o sursă de la care poate porni foarte ușor un atac.
+Folosind `objdump` inspectați funcția `main` a executabilului.
+
+> **TIP** Pentru a inspecta sursa, folosiți următoarea comandă:
+> ```Assembly
```

Comment:
Aici ar trebui să fie `Bash` în loc de `Assembly` (se vede syntax highlight-ul greșit în readme).

Reply:
Done